ui/src/app/guards/disabled-guard.service.ts

```diff
@@ -1,16 +1,24 @@
 import {Injectable} from '@angular/core';
 import {ActivatedRouteSnapshot, CanActivate, CanActivateChild, Router, RouterStateSnapshot} from '@angular/router';
 
+import {SignInService} from 'app/services/sign-in.service';
 import {profileApi} from 'app/services/swagger-fetch-clients';
 import {convertAPIError} from 'app/utils/errors';
 import {ErrorCode} from 'generated/fetch';
 
 @Injectable()
 export class DisabledGuard implements CanActivate, CanActivateChild {
-  constructor(private router: Router) {}
+  constructor(private router: Router, private signInService: SignInService) {}
 
   async canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Promise<boolean> {
     try {
+      // The user is not necessarily authenticated at this point - we have to
```

Comment:
You just have read my mind – I was adding a review comment asking for a comment, then your comment showed up when I refreshed :)

I'm still a little confused as to how we have confidence that this guard will be called again once the sign-in state is changed. Is this something we handle explicitly somewhere?

Reply:
The observable here blocks until the state is known, at which point it fires the first event which gets captured by this `toPromise`. Subsequently, it may fire again I believe if the user sign in state changes again, but we're no longer listening at that point.

Reply:
On initial sign in - we're currently dependent on a full page reload. Therefore signed out -> signed in is not a valid state transition anyways.